### DISCUSSION TITLE
Chart size

### DISCUSSION THREAD

**Discussion by juhopekka-xrtc** [patrykandpatrick/vico#936] [Questions] [answered]
### Question

Hi, it seems that the `CartesianChartHostBox` overrides it's modifier to always have `.height(CHART_HEIGHT.dp).fillMaxWidth()` where `CHART_HEIGHT` is 200 dp, or have I missed something?
 
Is there any way to change the height of the chart?

### Vico version(s)

2.0.0-beta.2

### UI framework(s)

Jetpack Compose

**Comment by patrickmichalik** [comment, marked as answer]
Hello! Unlike `Modifier.requiredHeight`, `Modifier.height` respects the incoming constraints. Here, it’s `a` that takes precedence, not `b`:

```kt
Modifier.height(a).height(b)
```

Thus, the code in question merely sets the default height 200 dp. This doesn’t preclude height customization—you can use `Modifier.height` as normal with `CartesianChartHost` to resize a chart.

**Reply by juhopekka-xrtc** [reply]
Oh yes, you are right, that's what happens with layout modifiers. There was a gap in my knowledge of how the layout constrains are passed in layout phase. That gap is now filled, thanks.

The height not being set was due to some other reason obviously, which ... of course doesn't reproduce anymore.

**Reply by patrickmichalik** [reply]
You’re welcome! Should the problem reoccur, feel free to reach out.
